compiler/parser/parser.cpp

```diff
@@ -350,6 +350,38 @@ static void printModuleSearchPath();
 static void helpPrintPath(Vec<const char*> path);
 static void ensureRequiredStandardModulesAreParsed();
 
+
+static void parseChplSourceFile(const char* inputFileName) {
+  /*
+    The selection of 16 here was chosen to provide enough space for
+    generating files like .tmp.obj (whose length is 8) from the
+    input filename; we doubled the value to ensure some breathing
+    room, and under an assumption that most files won't be this long
+    anyway.
+  */
+  const size_t reductionMaxLength = 16;
+  /*
+    Ensure that all the files parsed don't exceed
+    (NAME_MAX - reductionMaxLength) e.g. 239 bytes on
+    unix and linux system.
+  */
+  const size_t maxFileName = NAME_MAX - reductionMaxLength;
+  const char* baseName = stripdirectories(inputFileName);
+  if (strlen(baseName) > maxFileName)
+    {
```

Comment:
weird indenting

Reply:
I'm not normally a "curly bracket on its own line" kind of guy, but that's how the code was formatted before I refactored it and this is how emacs indents that style by default.  Would you like me to change this?  Do you care what I change it to?  (e.g., I'd probably change it to the curly bracket being on the previous line).

Reply:
(I assume the rationale is "we indent the single statement that follows a single-statement conditional and then we indent statements within a compound statement" and checking Wikipedia, it seems like this is the case and that it is "GNU style", which is the default for emacs.  I'm not advocating for preserving it per se, just explaining how it ended up like this and still wondering whether you want it changed).

Reply:
It's not a big deal but I'd change it. Throughout the compiler source code, things tend to be either

```
if (bla) {
  stmts;
}
```

or 

```
if (bla)
{
  stmts;
}
```

I consider both of the above typical for our code and totally acceptable.

I don't know of anybody on our team who wants to advocate for the `{` to be indented (which is what we have here). I'm sure there are people in the broader world who like that. 

But obviously it does not make much difference.